### ISSUE TITLE
dataclasses.get_most_energetic_cascade returns cascade with lower energy than dataclasses.get_most_energetic_inice_cascade (Trac #1030)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1024]
When using dataclasses.get_most_energetic_cascade, it can happen that the cascade returned by the method has a lower energy than the one returned by dataclasses.get_most_energetic_inice_cascade.

An example file where this happens, e.g. for the first event, can be found at /data/user/hbretz/cascades_vs_inice_cascades.i3

```text
In [43]: dataclasses.get_most_energetic_inice_cascade(tree).energy
Out[43]: 9226.722570765558

In [44]: dataclasses.get_most_energetic_cascade(tree).energy
Out[44]: 16.27508624107153
```

This issue occured in icerec trunk, r133025.

Migrated from https://code.icecube.wisc.edu/ticket/1030
```json
{
    "status": "closed", 
    "changetime": "2015-08-10T20:31:17", 
    "description": "When using dataclasses.get_most_energetic_cascade, it can happen that the cascade returned by the method has a lower energy than the one returned by dataclasses.get_most_energetic_inice_cascade.\n\nAn example file where this happens, e.g. for the first event, can be found at /data/user/hbretz/cascades_vs_inice_cascades.i3\n\n\n{{{\nIn [43]: dataclasses.get_most_energetic_inice_cascade(tree).energy\nOut[43]: 9226.722570765558\n\nIn [44]: dataclasses.get_most_energetic_cascade(tree).energy\nOut[44]: 16.27508624107153\n}}}\n\nThis issue occured in icerec trunk, r133025.\n", 
    "reporter": "hbretz", 
    "cc": "david.schultz@icecube.wisc.edu", 
    "resolution": "fixed", 
    "_ts": "1439238677802127", 
    "component": "combo core", 
    "summary": "dataclasses.get_most_energetic_cascade returns cascade with lower energy than dataclasses.get_most_energetic_inice_cascade", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2015-06-24T15:41:58", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150709T22:02:28: 
* david.schultz changed owner from "" to "olivas"
* david.schultz changed component from "cmake" to "icecube offline"
* david.schultz changed cc from "" to "david.schultz@icecube.wisc.edu"
* david.schultz changed status from "new" to "assigned"
* david.schultz changed milestone from "" to "offline V15-07-00"

**Comment by icecube-trac** [comment]
Trac update at 20150709T22:47:38: olivas changed attachment from "" to "test_get_most_energetic_cascade.py"

**Comment by icecube-trac** [comment]
Trac update at 20150709T22:48:15: olivas commented:

> Weird.  I can verify the results shown above for that I3File, but can't seem to write a failing test just yet. 
> 
> Attached is a passing test that I would have expected to fail.  I'll keep refining that test to try to isolate the problem.
>

**Comment by icecube-trac** [comment]
Trac update at 20150721T20:00:47: 
* david.schultz commented:

> In changeset 135414:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="135414"
> NaN has a way of messing things up in strange places. Compare more defensively.  And add a test case that fails before this change. fixes #1024
> ```


* david.schultz changed resolution from "" to "fixed"
* david.schultz changed status from "assigned" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20150810T20:31:17: 
* olivas changed milestone from "offline V15-07-00" to ""
* olivas commented:

> Milestone offline V15-07-00 deleted
